Update Server.java

dimits-ts opened the pull request.

Commit:
Update Server.java

Commit:
Update GameServer.java

Commit:
Update ChatServer.java

Commit:
Update GameEngine.java

Commit:
Update GameUI.java

Commit:
Update Server.java

Commit:
Update GameServer.java

Commit:
Update GameEngine.java

Commit:
Update ChatServer.java

Commit:
Update ChatServer.java

Commit:
Update Logging.java

Commit:
Update Server.java

Commit:
Update GameServer.java

Commit:
Update GameEngine.java

AAAlex-123 commented on Server.java:
```diff
@@ -66,14 +64,30 @@ public Server() {
+	 * @param boolean printStackTrace whether to show detailed crash reports
+	 * @see Server#getServerOptions() getServerOptions
+	 */
+	public Server(int playerCount,boolean printStackTrace) {
+		Server.printStackTrace = printStackTrace;
+		inputs = new ObjectInputStream[playerCount];
+		outputs = new ObjectOutputStream[playerCount];
+		symbols = new char[playerCount];
+	}
```
πού τα βρίσκει τα playerCount, printStackTrace;

AAAlex-123 commented on Server.java:
```diff
@@ -66,14 +64,30 @@ public Server() {
 	 */
-	protected abstract void run();
+	@Override
+	public abstract void run();
```
@override από το runnable interface που έχει run()?

AAAlex-123 commented on Server.java:
```diff
@@ -26,7 +27,7 @@
  * Abstract class to run a server; GameServer and ChatServer inherit from it.
  */
-public abstract class Server implements Logging {
+public abstract class Server implements Logging,Runnable {
```
γτ κάνει implement runnable η υπερκλάση; αφού κ η chatserver το κάνει κ η υπερκλαση δεν ασχολείται με run()

AAAlex-123 commented on ChatServer.java:
```diff
@@ -6,12 +6,14 @@
  * Server-side application to handle chat between players.
  */
-public class ChatServer extends Server {
+public class ChatServer extends Server implements Runnable{
```
γτ implements runnable? αφού κ η υπερκλάση το κάνει

AAAlex-123 commented on ChatServer.java:
```diff
@@ -37,6 +39,19 @@ public ChatServer() {
+	 * @param int playerCount number of connections
+	 * @see Server#Server() Server()
+	 */
+	public ChatServer(int playerCount,boolean printStackTrace) {
```
που τα βρίσκει τα playerCount, printStackTrace

AAAlex-123 commented on ChatServer.java:
```diff
@@ -119,7 +135,8 @@ protected void getConnections() {
 
 				log("\nChat Connection #%d established with '%c'", index, symbols[index]);
```
το log αυτό δουλεύει με την παλιά έκδοση `log(String msg, Object... args)`

AAAlex-123 commented on GameServer.java:
```diff
@@ -35,6 +38,27 @@ public GameServer() {
+	 */
+	public static void main(String[] args) {
+		GameServer server = new GameServer();
+		ChatServer chatServer = new ChatServer(playerCount,printStackTrace);
```
πού τα βρίσκει τα playerCount, printStackTrace?

Commit:
Update Logging.java

updated loggerr

kill me

Commit:
Update Logging.java

AAAlex-123 committed:
playerLabel now updates

added `public updateXXXXConnectionCounter(int)` to update the playerLabel JLabel when a player joins/leaves the chat/game servers

AAAlex-123 committed:
added methods to update label

see Server commit

AAAlex-123 committed:
server now changes screen, logerr fix

Commit:
Update ChatServer.java

Commit:
Update Logging.java

Commit:
Update GameEngine.java

Commit:
Update GameServer.java

AAAlex-123 reviewed (approved):
```n i c e```

AAAlex-123 committed:
Merge pull request #25 from AAAlex-123/Server-problem-2-Crash-harder

Update ChatServer.java
